app/controllers/samples_controller.rb

```diff
@@ -495,15 +495,18 @@ def update
   # DELETE /samples/1
   # DELETE /samples/1.json
   def destroy
-    deletable = @sample.pipeline_runs.empty?
+    # Will also delete from job_stats, ercc_counts, backgrounds_pipeline_runs, pipeline_runs, input_files, and backgrounds_samples
+    # Current job_status values look like "1.Host Filtering-FAILED" or "2.GSNAPL/RAPSEARCH alignment-FAILED"
+    # Allow deletion if pipeline runs are empty or the latest run failed, and the sample belongs to you, or you're an admin.
+    deletable = ((@sample.pipeline_runs.empty? || @sample.pipeline_runs[0].job_status.include?("FAILED")) && @sample.user_id == current_user.id) || current_user.admin?
```

Comment:
What if postprocessing failed but the report is ready?
This is probably complex enough to warrant its own function like https://github.com/chanzuckerberg/idseq-web/blob/master/app/models/sample.rb#L381

Reply:
Hm, ok